code/datums/components/edible.dm

```diff
@@ -0,0 +1,268 @@
+/datum/component/edible/proc/TakeBite(mob/living/eater, mob/living/feeder)
+	if(!owner.reagents)
+		return FALSE
+	if(eater.satiety > -200)
+		eater.satiety -= junkiness
+	playsound(eater.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	if(owner.reagents.total_volume)
+		SEND_SIGNAL(parent, COMSIG_FOOD_EATEN, eater, feeder)
+		var/fraction = min(bite_consumption / owner.reagents.total_volume, 1)
+		owner.reagents.trans_to(eater, bite_consumption, transfered_by = feeder, method = INGEST)
+		bitecount++
+		On_Consume(eater)
+		checkLiked(fraction, eater)
+
+		//Invoke our after eat callback if it is valid
+		if(after_eat)
+			after_eat.Invoke(eater, feeder)
+
+		return TRUE
+
+///Checks whether or not the eater can actually consume the food
+/datum/component/edible/proc/CanConsume(mob/living/eater, mob/living/feeder)
+	if(!iscarbon(eater))
+		return FALSE
+	var/mob/living/carbon/C = eater
+	var/covered = ""
+	if(C.is_mouth_covered(head_only = 1))
+		covered = "headgear"
+	else if(C.is_mouth_covered(mask_only = 1))
+		covered = "mask"
+	if(covered)
+		var/who = (isnull(feeder) || eater == feeder) ? "your" : "[eater.p_their()]"
+		to_chat(feeder, "<span class='warning'>You have to remove [who] [covered] first!</span>")
+		return FALSE
+	return TRUE
+
+///Check foodtypes to see if we should send a moodlet
+/datum/component/edible/proc/checkLiked(var/fraction, mob/M)
+	if(last_check_time + 50 > world.time)
+		return FALSE
+	if(!ishuman(M))
+		return FALSE
+	var/mob/living/carbon/human/H = M
+	if(HAS_TRAIT(H, TRAIT_AGEUSIA) && foodtypes & H.dna.species.toxic_food)
+		to_chat(H, "<span class='warning'>You don't feel so good...</span>")
+		H.adjust_disgust(25 + 30 * fraction)
+	else
+		if(foodtypes & H.dna.species.toxic_food)
+			to_chat(H,"<span class='warning'>What the hell was that thing?!</span>")
+			H.adjust_disgust(25 + 30 * fraction)
+			SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "toxic_food", /datum/mood_event/disgusting_food)
+		else if(foodtypes & H.dna.species.disliked_food)
+			to_chat(H,"<span class='notice'>That didn't taste very good...</span>")
+			H.adjust_disgust(11 + 15 * fraction)
+			SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "gross_food", /datum/mood_event/gross_food)
+		else if(foodtypes & H.dna.species.liked_food)
+			to_chat(H,"<span class='notice'>I love this taste!</span>")
+			H.adjust_disgust(-5 + -2.5 * fraction)
+			SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "fav_food", /datum/mood_event/favorite_food)
+	if((foodtypes & BREAKFAST) && world.time - SSticker.round_start_time < STOP_SERVING_BREAKFAST)
+		SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "breakfast", /datum/mood_event/breakfast)
+	last_check_time = world.time
+
+///Delete the item when it is fully eaten
+/datum/component/edible/proc/On_Consume(mob/living/eater)
+	if(!eater)
+		return
+	if(!owner.reagents.total_volume)
+		qdel(parent)
+
+///Ability to feed food to puppers
+/datum/component/edible/proc/UseByAnimal(datum/source, mob/user)
+	if(!isdog(user))

```

Comment:
shouldn't you make a subtype that handles this bit by only being added to the corgi mobs?

Reply:
I might be misunderstanding, but this component is for the object that is eaten, so this means that if you use an item on an animal; this code is run. all foods can (in theory) be eaten by dogs at the moment.